druid/src/widget/svg.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! An SVG widget.
+
+use crate::{
+    kurbo::BezPath,
+    piet::{Color, RenderContext},
+    BaseState, BoxConstraints, Data, Env, Event, EventCtx, LayoutCtx, PaintCtx, Point, Size,
+    UpdateCtx, Widget,
+};
+
+use log::error;
+use std::error::Error;
+use std::marker::PhantomData;
+use std::str::FromStr;
+
+use usvg;
+
+/// A widget that renders a SVG
+pub struct Svg<T> {
+    // On construction the SVG string is parsed into a SvgData.
+    svg_data: SvgData,
+    phantom: PhantomData<T>,
+}
+
+impl<T: Data> Svg<T> {
+    /// Create an SVG-drawing widget from a valid SVG string literal.
+    ///
+    /// The SVG will scale to fit its box constraints.
+    /// If SVG is invalid a blank SVG will be rendered instead.
+    pub fn new(svg_str: &str) -> impl Widget<T> {
+        let svg_data = SvgData::from_str(svg_str);
```

Comment:
you know the trick with `FromStr`? that it gives you a new fn on string, `parse()`, so this could also be written,

```rust
let svg_data = svg_str.parse::<SvgData>();
```

(this isn't a suggested change, I'm just pointing out this feature)